src/nupic/algorithms/SDRClassifier.cpp

```diff
@@ -185,35 +186,42 @@ void SDRClassifier::compute(UInt recordNum, const vector<UInt> &patternNZ,
 
 void SDRClassifier::infer_(const vector<UInt> &patternNZ,
                            const vector<Real64> &actValue,
-                           ClassifierResult &result) {
-  // add the actual values to the return value. For buckets that haven't
+                           ClassifierResult &result)
+{
+  // Add the actual values to the return value. For buckets that haven't
   // been seen yet, the actual value doesn't matter since it will have
   // zero likelihood.
-  vector<Real64> *actValueVector =
-      result.createVector(-1, (UInt)actualValues_.size(), 0.0);
-  for (UInt i = 0; i < (UInt)actualValues_.size(); ++i) {
+  vector<Real64> &actValueVector = result[-1];
```

Comment:
//what is `map[-1]` ?

Reply:
..understood below. Either use the `const Int ACTUAL_STEP = -1;` 
or //add a comment here